src/Workspaces/CoreTest/SolutionTests/SolutionWithSourceGeneratorTests.cs

```diff
@@ -779,5 +779,43 @@ public async Task DynamicFilesNotPassedToSourceGenerators()
             Assert.True(noTreesPassed.HasValue);
             Assert.False(noTreesPassed!.Value);
         }
+
+        [Fact]
+        public async Task LinkedDocumentOfFrozenShouldNotRunSourceGenerator()
```

Comment:
Slight nit: move this up by ForkAfterFreezeNoLongerRunsGenerators where the other tests are around frozen behavior.